sdks/python/apache_beam/runners/portability/stager_test.py

```diff
@@ -420,134 +375,167 @@ def test_sdk_location_local_directory_not_present(self):
     sdk_location = 'nosuchdir'
     with self.assertRaises(RuntimeError) as cm:
       options = PipelineOptions()
-      options.view_as(GoogleCloudOptions).staging_location = staging_dir
       self.update_options(options)
       options.view_as(SetupOptions).sdk_location = sdk_location
 
-      dependency.stage_job_resources(options)
+      self.stager.stage_job_resources(options, staging_location=staging_dir)
     self.assertEqual(
         'The file "%s" cannot be found. Its '
         'location was specified by the --sdk_location command-line option.' %
-        sdk_location,
-        cm.exception.args[0])
+        sdk_location, cm.exception.args[0])
 
-  def test_sdk_location_gcs_source_file(self):
+  def test_sdk_location_remote_source_file(self):
     staging_dir = self.make_temp_dir()
     sdk_location = 'gs://my-gcs-bucket/tarball.tar.gz'
 
     options = PipelineOptions()
-    options.view_as(GoogleCloudOptions).staging_location = staging_dir
     self.update_options(options)
     options.view_as(SetupOptions).sdk_location = sdk_location
 
-    with mock.patch('apache_beam.runners.dataflow.internal.'
-                    'dependency._dependency_file_copy'):
-      self.assertEqual(
-          [names.DATAFLOW_SDK_TARBALL_FILE],
-          dependency.stage_job_resources(options))
-
-  def test_sdk_location_gcs_wheel_file(self):
+    with mock.patch('.'.join([
+        self.__module__, TestStager.__name__, TestStager.stage_artifact.__name__
+    ])):
+      with mock.patch('.'.join([
+          self.__module__, TestStager.__name__,
+          TestStager._download_file.__name__
+      ])):
+        self.assertEqual([names.DATAFLOW_SDK_TARBALL_FILE],
+                         self.stager.stage_job_resources(
+                             options, staging_location=staging_dir))
+
+  def test_sdk_location_remote_wheel_file(self):
     staging_dir = self.make_temp_dir()
     sdk_filename = 'apache_beam-1.0.0-cp27-cp27mu-manylinux1_x86_64.whl'
-    sdk_location = 'gs://my-gcs-bucket/' + sdk_filename
+    sdk_location = '/tmp/remote/my-bucket/' + sdk_filename
 
     options = PipelineOptions()
-    options.view_as(GoogleCloudOptions).staging_location = staging_dir
     self.update_options(options)
     options.view_as(SetupOptions).sdk_location = sdk_location
 
-    with mock.patch('apache_beam.runners.dataflow.internal.'
-                    'dependency._dependency_file_copy'):
-      self.assertEqual(
-          [sdk_filename],
-          dependency.stage_job_resources(options))
+    # We can not rely on actual remote file systems paths hence making
+    # '/tmp/remote/' a new remote path.
+    def is_remote_path(dummy_self, path):
```

Comment:
Are you sure we need to do this?

Reply:
yes, as I don't see other ways to spoof remote path.